main.py

```diff
@@ -1,10 +1,36 @@
-"""This file will use all of our functions and run the progam"""
+"""This file will use all of our functions and run the program"""
 
+import time
+import typer
+
+# Import from configator
 from configator import create_actions_config
 from configator import create_gatorgradle_yml
 from configator import generate_build_gradle
 
-if __name__ == '__main__':
+app = typer.Typer()
+@app.command()
+def main(
+    name: str = typer.Option("configator-generated"),
+    brk: str = typer.Option("true"),
+    fastfail: str = typer.Option("false"),
+    ind: int = typer.Option(2),
+    vers: str = typer.Option("v1.1.0"),
+    ggradleversion: str = typer.Option("0.5.1"),
+    com: int = typer.Option(3)
+):
+    # Alert user of assignment generation under specified name
+    typer.echo(f"\nGenerating assignment as: {name}\n")
+    # Progress bar for visual appeal
+    with typer.progressbar(range(100)) as progress:
+        for value in progress:
+            # Fake processing time
+            time.sleep(0.01)
```

Comment:
Instead of artificially creating lag here, can the progress bar be tied to the execution of the functions that generate the files?

Reply:
This should definitely not be artificially simulated; progress bars can be very annoying if they don't actually mark any progress, especially when they take an entire second to pass. If you'd like to keep the progress bar but make it actually track real progress, look at the [iterate manually](https://typer.tiangolo.com/tutorial/progressbar/#iterate-manually) section of the Typer documention. Moving your function calls inside the `with` statement (and removing the loop), and then calling `progress.update(<value>)`  after each function call should net you a progress bar that jumps up after each function completes; it likely will finish extremely quickly, but that is a good thing!